src/app/services/post.service.ts

```diff
@@ -0,0 +1,24 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {Post} from "../post.model";
+
+@Injectable({
+    providedIn: 'root'
+})
+export class PostService {
+    private postsUrl = 'http://localhost:8080/post';
```

Comment:
This way of providing the URL is not appropriate. The accepted way of providing a `baseUrl` into the HttpClient (that is, `http://localhost:8080`) is by using a HTTP interceptor.

We will need the interceptor anyway to provide authentication capabilities. It's okay if it is added in a future PR.